Jenkinsfile

```diff
@@ -574,27 +587,36 @@ pipeline {
                 stage('Fp32 Hip /opt/rocm') {
                     when {
                         beforeAgent true
-                        expression { params.TARGET_VEGA20 || params.TARGET_VEGA10 }
+                        expression { params.TARGET_VEGA20 || params.TARGET_VEGA10 || params.TARGET_GFX908 }
```

Comment:
The original motivation is resource allocation in CI pool, somehow vega nodes become bottleneck and gfx908 nodes are stable and have a few. 
I think ASIC specific tests should be in "full-test" while smoke tests could be "anyGPU"? well maybe not all of them but the kinds which we have more than a few in CI pools.

Reply:
@junliume Note that this breaks the intent of things like `params.TARGET_GFX908`. The idea is to limit testing to the specific targets. E.g. if I check only TARGET_GFX908, then only gfx908-specific tests will be run and only on gfx908 nodes. After this change this is not so anymore. For example, if I want to run only gfx908 tests, this stage may run on Vega node, depending on which node is free.

Now we have an `AnyGPU` that can run on any GPU, no matter which `params.TARGET_*` is checked. The programmer can no longer rely on running a particular branch on a particular type of GPU.

Fortunately, this problem only applies to smoke tests.

[Notice] Before this change (and due to historical reasons), this kind of `||` machinery was used only for Vega nodes (`params.TARGET_VEGA20 || params.TARGET_VEGA10`), which was not good, but tolerable, because these nodes are very similar.

[Proposal]
If we want to keep this AnyGPU stuff then I recommend making this more explicit:
- Introduce the "AnyGPU" label and set it on the nodes that we want to be used in "any GPU" tests
- introduce `params.TARGET_ANYGPU` in Jenkinsfile and apply it instead of `... || ...` constructs in relevant stages. At the same time, use `label rocmnode("AnyGPU")` in the same stages.

This way, when the programmer wants to run a custom task, `TARGET_ANYGPU` will be visible and one can explicitly decide whether to run GPU-independent tests or not.

/CC @shurale-nkn @JehandadKhan

Reply:
Lets wait for @nyamodil 's PR to remove the smoke test and then address this issue.